src/Component.js

```diff
@@ -1,48 +1,52 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import debounce from 'lodash.debounce';
-import {shouldComponentUpdate} from 'react/lib/ReactComponentWithPureRenderMixin';
-
-
-export const DebounceInput = React.createClass({
-  propTypes: {
-    element: React.PropTypes.oneOfType([React.PropTypes.string, React.PropTypes.func]),
-    type: React.PropTypes.string,
-    onChange: React.PropTypes.func.isRequired,
-    onKeyDown: React.PropTypes.func,
-    onBlur: React.PropTypes.func,
-    value: React.PropTypes.oneOfType([
-      React.PropTypes.string,
-      React.PropTypes.number
+
+
+export class DebounceInput extends React.PureComponent {
+  static propTypes = {
+    element: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+    type: PropTypes.string,
+    onChange: PropTypes.func.isRequired,
+    onKeyDown: PropTypes.func,
+    onBlur: PropTypes.func,
+    value: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number
     ]),
-    minLength: React.PropTypes.number,
-    debounceTimeout: React.PropTypes.number,
-    forceNotifyByEnter: React.PropTypes.bool,
-    forceNotifyOnBlur: React.PropTypes.bool
-  },
-
-
-  getDefaultProps() {
-    return {
-      element: 'input',
-      type: 'text',
-      minLength: 0,
-      debounceTimeout: 100,
-      forceNotifyByEnter: true,
-      forceNotifyOnBlur: true
-    };
-  },
+    minLength: PropTypes.number,
+    debounceTimeout: PropTypes.number,
+    forceNotifyByEnter: PropTypes.bool,
+    forceNotifyOnBlur: PropTypes.bool
+  };
+
+
+  static defaultProps = {
+    element: 'input',
+    type: 'text',
+    minLength: 0,
+    debounceTimeout: 100,
+    forceNotifyByEnter: true,
+    forceNotifyOnBlur: true
+  };
 
 
-  getInitialState() {
-    return {
-      value: this.props.value || ''
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: props.value || ''
     };
-  },
+
+    this.isDebouncing = false;
+
+    this.onChange = this.onChange.bind(this);
```

Comment:
Since you've added `transform-class-properties` we can use lambdas instead of instance methods like:

```js
  //...
  onChange = event => 
    this.notify({...event, target: {...event.target, value: ''}});
```